### ISSUE TITLE
can't save to a folder mount

### ISSUE THREAD

**Issue by checktext00** [joncampbell123/dosbox-x#5313]
### Describe the bug

I'm using Windows 95 C (OSR 2.5) with DOSBox-X SDL1 and using the [win95.conf from the guide](https://dosbox-x.com/wiki/Guide%3AInstalling-Windows-95#_dosbox_x_config_file). When I added to the win95.conf and external folder mount like so:
```
[autoexec]
IMGMOUNT 0 empty -fs none -t floppy
IMGMOUNT 1 empty -fs none -t floppy -size 512,15,2,80
IMGMOUNT C win95.img -ide 1m
**MOUNT D win95foldermount -ide 1m**
IMGMOUNT E win95.iso -t iso -ide 2m
BOOT C:
```
it won't save my changes on the host OS, meaning when I add a file from Windows 95 to the D drive, it seems to be correctly saved, but on my host OS filesystem it's not there, and when I quit and restart Windows 95 the file is not there (but the files I add from my host OS correctly show up in Windows 95)

I **don't** have `convertdrivefat = true` or `convert fat free space=0` added under `[sdl]`

Also, all the options under the DOSBox-X "Drive" menu are greyed-out under all drive letters, except drive letter D has a "Save to disk image" option available. Shouldn't there be options to mount a drive for the other available letters F through Z?

### Steps to reproduce the behaviour

create a folder on the host OS with some stuff in it, launch Windows 95 C (OSR 2.5) with the [win95.conf from the DOSBox-X guide](https://dosbox-x.com/wiki/Guide%3AInstalling-Windows-95#_dosbox_x_config_file), add that folder as a mount like so:
```
[autoexec]
MOUNT D win95foldermount -ide 1m
```

see that you can access the files from Windows 95, and edit them from Windows 95, but the changed aren't saved on the host OS at all.

Also from the DOSBox-X menu go to Drive, then choose any letter and see that everything is greyed out

### Expected behavior

The folder mount should be writable and saved on the host OS, the Drive menu should allow mounting drives while Windows 95 is running

### What operating system(s) this bug have occurred on?

macOS 11

### What version(s) of DOSBox-X have this bug?

2024.12.04 (latest stable)

### Additional information

Thanks for looking into it

### Have you checked that no similar bug report(s) exist?

- [X] I have searched and didn't find any similar bug report.

### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

**Comment by rderooy** [comment]
Correct, when using folder mounts with a guest OS, any changes to the folder mount are lost when you shutdown the guest OS. This is documented as such on the wiki.

**Comment by joncampbell123** [comment]
The fake FAT filesystem generated to present files to Windows 95 doesn't reflect changes back to the host.
